src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AmendReferralService.kt

```diff
@@ -57,4 +58,50 @@ class AmendReferralService(
     return referralService.getSentReferralForUser(id, user)
       ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "sent referral not found [id=$id]")
   }
+
+  fun updateNeedAndRequirement(referral: Referral, update: AmendNeedAndRequirementDTO, authentication: JwtAuthenticationToken): AmendNeedAndRequirementDTO? {
+    val actor = userMapper.fromToken(authentication)
+    val oldValue = ReferralAmendmentDetails(
+      furtherInformation = referral.furtherInformation,
+      additionalNeedsInformation = referral.additionalNeedsInformation,
+      accessibilityNeeds = referral.accessibilityNeeds,
+      needsInterpreter = referral.needsInterpreter,
+      interpreterLanguage = referral.interpreterLanguage,
+      hasAdditionalResponsibilities = referral.hasAdditionalResponsibilities,
+      whenUnavailable = referral.whenUnavailable
+    )
+    val newValue = ReferralAmendmentDetails(
+      additionalNeedsInformation = update.additionalNeedsInformation,
+      furtherInformation = update.furtherInformation,
+      accessibilityNeeds = update.accessibilityNeeds,
+      needsInterpreter = update.needsInterpreter,
+      interpreterLanguage = update.interpreterLanguage,
+      hasAdditionalResponsibilities = update.hasAdditionalResponsibilities,
+      whenUnavailable = update.whenUnavailable
+
+    )
+    referral.additionalNeedsInformation = update.additionalNeedsInformation
+    referral.accessibilityNeeds = update.accessibilityNeeds
+    referral.needsInterpreter = update.needsInterpreter
+    referral.interpreterLanguage = update.interpreterLanguage
+    referral.hasAdditionalResponsibilities = update.hasAdditionalResponsibilities
+    referral.whenUnavailable = update.whenUnavailable
+    referral.furtherInformation = update.furtherInformation
+
+    val changelog = Changelog(
+      referral.id,
+      UUID.randomUUID(),
+      "NEED_AND_REQUIREMENT",
+      oldValue,
+      newValue,
```

Comment:
🙅 _Technical issue_: We cannot serialise DTOs into the JSON fields because it would create a schema dependency

🙅 I think the team decided to serialise the values as arrays, which requires us to store each field change independently, with a separate reason for the change.

E.g. `referral_id, UUID, "additionalNeedsInformation", [old], [new], ...`